### ISSUE TITLE
Failed saving animation when using napari.utils.DirectLabelColormap

### ISSUE THREAD

**Issue by manerotoni** [napari/napari-animation#226]
Hello,
with napari-animation and I encountered a reproducible error when using  `DirectLabelColormap`. Such a map is needed as I do class-labelling and it allows easy configuration and consistency in the colors. 

I add two small code snipptes. The first save the data correctly. In the  second example the `animate` command fails.

```
# Succesful example
from napari import Viewer
from napari_animation import Animation
import numpy as np
from napari.utils import DirectLabelColormap

viewer = Viewer()
viewer.add_labels(test_lbl)
animation = Animation(viewer)
out_name_mov = './test.mov'
viewer.add_labels(test_lbl)
for idx in range(0,10):
    viewer.dims.current_step = (idx,4, 4)
    animation.capture_keyframe()
animation.animate(out_name_mov, canvas_only=True, quality = 1)
```

```
# Failed example when using a colormap DirectLabelColormap()
cmap = DirectLabelColormap()
cmap.color_dict = {None: None, 0: None, 1:'blue', 2:'red'}

viewer = Viewer()
viewer.add_labels(test_lbl, colormap=cmap)

animation = Animation(viewer)
out_name_mov = './test.mov'
viewer.add_labels(test_lbl)
for idx in range(0,10):
    viewer.dims.current_step = (idx,4, 4)
    animation.capture_keyframe()

# This command fails with TypeError: cannot pickle '_nrt_python._MemInfo' object
animation.animate(out_name_mov, canvas_only=True, quality = 1)
```

**Comment by brisvag** [comment]
Might be caused by the recent napari/napari#7025...

**Comment by jni** [comment]
> Might be caused by the recent

no, based on [this SO question](https://stackoverflow.com/questions/58000229/why-a-dict-type-variable-content-cannot-be-pickled) it's because there is a numba typed dict in the class, so the class cannot be pickled. I think the solution would be to add a `__pickle__` or `__copy__` method (I'm not sure what the right magic dunder methods are here) to DirectLabelColormap that will swap out the numba dict for a regular dict.

There *might* be a user-space solution here (check if it's a DirectLabelColormap, replace it with a dict — thanks to napari/napari#7025, actually. 😂), so I'll leave this issue here and make a more targeted one in napari. Thanks @manerotoni for the report! 🙏

**Comment by psobolewskiPhD** [comment]
I think this came up with the PRs fixing some of the state/properties comparisons.
Does interpolation of colors of labels even make sense -- particularly for a user provided direct colormap?

**Comment by jni** [comment]
You might not want to interpolate the colors, (in fact, in the case of colormapped values, you *certainly* don't want to interpolate in color space), but that can be done by setting interpolation=None. I can see it being useful to change the colormap in the middle of an animation, so *checkpointing* a colormap is certainly a necessary feature.

**Comment by Czaki** [comment]
> no, based on [this SO question](https://stackoverflow.com/questions/58000229/why-a-dict-type-variable-content-cannot-be-pickled) it's because there is a numba typed dict in the class, so the class cannot be pickled. I think the solution would be to add a `__pickle__` or `__copy__` method (I'm not sure what the right magic dunder methods are here) to DirectLabelColormap that will swap out the numba dict for a regular dict.

I think that `__copy__` is enough.

**Comment by brisvag** [comment]
@psobolewskiPhD this is still happening after release?

**Comment by psobolewskiPhD** [comment]
There is still an issue with numba, which isn't present with a simple dev env here but is present with a real napari install
https://github.com/napari/napari-animation/pull/252#issuecomment-3970588884

From the convo above it seems like it needs to be fixed in napari/napari
